Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -158,11 +158,11 @@ class MessageDTO: NSManagedObject {
         return request
     }
 
-    private static func allAttachmentsAreUploadedOrEmptyPredicate() -> NSCompoundPredicate {
-        NSCompoundPredicate(orPredicateWithSubpredicates: [
-            .init(format: "NOT (ANY attachments.localStateRaw != %@)", LocalAttachmentState.uploaded.rawValue),
-            .init(format: "attachments.@count == 0")
-        ])
+    static func allAttachmentsAreUploadedOrEmptyPredicate() -> NSCompoundPredicate {
+        .init(
+            format: "SUBQUERY(attachments, $a, $a.localStateRaw == %@).@count == attachments.@count",
```

Comment:
Awesome! (Maybe it's helpful to start thinking adding FetchIndexes for fields like this one)

Reply:
I think in this case this will be a small and simple query, but if we see that adding fetch indexes would be benificial down the road we can add them 🙂